.github/workflows/release.yaml

```diff
@@ -166,3 +151,50 @@ jobs:
     #     TWINE_USERNAME=<USERNAME>
     #     TWINE_PASSWORD=<PASSWORD>
     #     TWINE_REPOSITORY_URL=<REPOSITORY_URL>
+
+  # Generate the build provenance. The generator should be referenced with a semantic version.
+  # The build will fail if we reference it using the commit SHA. To avoid using a pre-built
+  # provenance generator which depends on an external service Rekor (https://github.com/sigstore/rekor)
+  # we build this generator from source for now. For more information see this discussion:
+  # https://github.com/slsa-framework/slsa-github-generator/issues/942
+  provenance:
+    needs: [build, release]
+    uses: slsa-framework/slsa-github-generator/.github/workflows/generator_generic_slsa3.yml@v1.2.0
+    with:
+      base64-subjects: ${{ needs.build.outputs.artifacts-sha256 }}
+      compile-generator: true
+    permissions:
+      actions: read # To read the workflow path.
+      id-token: write # To sign the provenance.
+      contents: write # To add assets to a release.
+
+
+  # Publish the SLSA provenance as the GitHub release asset.
+  publish_provenance:
+    needs: provenance
+    name: Publish provenance
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write # To publish release notes.
+    steps:
+
+    - name: Harden Runner
+      uses: step-security/harden-runner@2e205a28d0e1da00c5f53b161f4067b052c61f34 # v1.5.0
+      with:
+        egress-policy: audit # TODO: change to 'egress-policy: block' after couple of runs
+
+    - name: Check out repository
+      uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b # v3.0.2
+      with:
+        fetch-depth: 0
+
+    - name: Download provenance
+      uses: actions/download-artifact@fb598a63ae348fa914e94cd0ff38f362e927b741 # v3.0.0
+      with:
+        name: ${{ needs.provenance.outputs.attestation-name }}
+
+    - name: Upload provenance
+      run: |
+        gh release upload `git describe --tags --abbrev=0` ${{ needs.provenance.outputs.attestation-name }}
```

Comment:
I wonder: if we output the result of `git describe --tags --abbrev=0` from the `release` job and read it here in the `publish_provenance` job, then we wouldn’t need to run the `checkout` action?

Also, it should be safe to write the attestation name into the shell command—we/SLSA folks own that name so there’s no injection issue here?

Reply:
> I wonder: if we output the result of `git describe --tags --abbrev=0` from the `release` job and read it here in the `publish_provenance` job, then we wouldn’t need to run the `checkout` action?

I think it's a good idea to pass the release tag between jobs, but unfortunately `gh` requires a git repo to work, so we can't avoid checking out the repo.

Reply:
> Also, it should be safe to write the attestation name into the shell command—we/SLSA folks own that name so there’s no injection issue here?

I'm assuming we can trust the attestation name...

Reply:
See commit [3b48cdb](https://github.com/jenstroeger/python-package-template/pull/342/commits/3b48cdb88b7aca55b89729b2c7bdde3bee8364e9).